File: src/main/java/no/jstien/bikeapi/tsdb/read/Query.java
package no.jstien.bikeapi.tsdb.read;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Query {
    private String aggregator = "sum";
    private String metric;
    private List<Filter> filters;
    private String downsample = "1s-sum";

    public Query(String metric) {
        this.metric = metric;
        this.filters = new ArrayList<>();
    }

    public void setAggregator(String aggregator) {
        this.aggregator = aggregator;
    }

    public void setDownsample(String downsample) {
        this.downsample = downsample;
    }

    public void addFilter(Filter filter) {
        filters.add(filter);
    }

    public List<Filter> getFilters() {
        return Collections.unmodifiableList(filters);
    }
}

File: src/main/java/no/jstien/bikeapi/tsdb/read/Query.kt
package no.jstien.bikeapi.tsdb.read

import java.util.ArrayList
import java.util.Collections

class Query(private val metric: String) {
    private var aggregator = "sum"
    private val filters: MutableList<Filter>
    private var downsample = "1s-sum"

    init {
        this.filters = ArrayList()
    }

    fun setAggregator(aggregator: String) {
        this.aggregator = aggregator
    }

    fun setDownsample(downsample: String) {
        this.downsample = downsample
    }

    fun addFilter(filter: Filter) {
        filters.add(filter)
    }

    fun getFilters(): List<Filter> {
        return Collections.unmodifiableList(filters)
    }
}
